ngx_http_vod_module.c

```diff
@@ -3418,11 +3422,19 @@ ngx_http_vod_async_http_read(ngx_http_vod_http_reader_state_t *state, ngx_buf_t
 	child_params.range_start = offset;
 	child_params.range_end = offset + size;
 
+	upstream_location = &conf->upstream_location;
+
+	if (conf->request_handler == ngx_http_vod_mapped_request_handler &&
+		conf->remote_upstream_location.len != 0)
+	{
+		upstream_location = &conf->remote_upstream_location;
+	}
+
```

Comment:
better add the upstream_location to ngx_http_vod_http_reader_state_t, and calculate it once in ngx_http_vod_http_reader_open_file. there you can just check if ctx->state == STATE_MAP_OPEN to know which location to use